packages/components/src/unit-control/utils.js

```diff
@@ -61,7 +61,7 @@ export function parseUnit( initialValue, units = CSS_UNITS ) {
 	unit = unit.toLowerCase();
 
 	if ( hasUnits( units ) ) {
-		const match = units.find( ( item ) => item.value === unit );
+		const match = keyBy( units, 'value' )[ unit ];
```

Comment:
@ItsJonQ: Ah, I wasn't clear, but I meant that a keyed version of `units` could be kept where it can be reused — so possibly as a module constant. Otherwise we're still iterating over the collection every time!

Reply:
@mcsf Ah, gotcha! It's a little tricky, as the units coming in may not be constant. Ultimately, this is determined by the `add_theme_support` config within the context of Gutenberg.

We could memoize it? I'm wondering if it's worth it, considering this operation is quite small (within the grand scheme of things)

Reply:
We could memoise, yeah, but I don't want to take more of your time on this now, so I suggest reverting to the initial array and `find` situation, and we can improve from there. :)

Reply:
Sounds good! Thank you @mcsf !!